### PR TITLE
Support for namespaces with qhub deployments on kubernetes clusters

### PR DESCRIPTION
**Pull request by costrouc**
Closes https://github.com/Quansight/qhub-cloud/issues/432

### CODE REVIEW

File: tests/test_schema.py [hunk 1]
```diff
@@ -5,18 +5,20 @@
...
-        ("gcp-pytest", "gcp.qhub.dev", "gcp", "github-actions", "github"),
+        ("do-pytest", "dev", "do.qhub.dev", "do", "github-actions", "github"),
+        ("aws-pytest", "dev", "aws.qhub.dev", "aws", "github-actions", "github"),
+        ("gcp-pytest", "dev", "gcp.qhub.dev", "gcp", "github-actions", "github"),
+        ("azure-pytest", "dev", "azure.qhub.dev", "azure", "github-actions", "github"),
     ],
 )
 def test_schema(
-    project, domain, cloud_provider, ci_provider, auth_provider
+        project, namespace, domain, cloud_provider, ci_provider, auth_provider
```
Comment (tylerpotts) [inline tests/test_schema.py:17]:
```suggestion
    project, namespace, domain, cloud_provider, ci_provider, auth_provider
```

**Review by tylerpotts — APPROVED** [review]
LGTM other than whitespacing suggestion

File: tests/test_schema.py (hunk 1 above)
Comment (costrouc) [reply tests/test_schema.py]: fixed.

**Comment by costrouc** [comment]
Got agreement from @aktech , @tylerpotts , and @dharhas on the namespace name. Make namespace optional. 

We may have to do some checks down the road with namespace naming of resources.

**Pull request merged**